graylog2-server/src/main/java/org/graylog2/indexer/results/FieldStatsResult.java

```diff
@@ -28,60 +29,67 @@
 public class FieldStatsResult extends IndexQueryResult {
 
     @Nullable
+    private final ValueCount valueCount;
+    @Nullable
     private final ExtendedStats extendedStats;
     @Nullable
     private final Cardinality cardinality;
     private List<ResultMessage> searchHits;
 
 
-    public FieldStatsResult(@Nullable ExtendedStats extendedStats,
+    public FieldStatsResult(@Nullable ValueCount valueCount,
+                            @Nullable ExtendedStats extendedStats,
                             @Nullable Cardinality cardinality,
                             SearchHits hits,
                             String query,
                             BytesReference source, TimeValue took) {
         super(query, source, took);
-
-
+        this.valueCount = valueCount;
         this.extendedStats = extendedStats;
         this.cardinality = cardinality;
         this.searchHits = buildResults(hits);
     }
 
     public long getCount() {
-        return extendedStats != null ? extendedStats.getCount() : 0;
+        if (valueCount != null) {
+            return valueCount.getValue();
+        } else if (extendedStats != null) {
+            return extendedStats.getCount();
+        }
+        return Long.MIN_VALUE;
     }
 
     public double getSum() {
-        return extendedStats != null ? extendedStats.getSum() : 0;
+        return extendedStats != null ? extendedStats.getSum() : Double.NaN;
     }
 
     public double getSumOfSquares() {
-        return extendedStats != null ? extendedStats.getSumOfSquares() : 0;
+        return extendedStats != null ? extendedStats.getSumOfSquares() : Double.NaN;
     }
 
     public double getMean() {
-        return extendedStats != null ? extendedStats.getAvg() : 0;
+        return extendedStats != null ? extendedStats.getAvg() : Double.NaN;
     }
 
     public double getMin() {
-        return extendedStats != null ? extendedStats.getMin() : 0;
+        return extendedStats != null ? extendedStats.getMin() : Double.NaN;
     }
 
     public double getMax() {
-        return extendedStats != null ? extendedStats.getMax() : 0;
+        return extendedStats != null ? extendedStats.getMax() : Double.NaN;
     }
 
     public double getVariance() {
-        return extendedStats != null ? extendedStats.getVariance() : 0;
+        return extendedStats != null ? extendedStats.getVariance() : Double.NaN;
     }
 
     public double getStdDeviation() {
-        return extendedStats != null ? extendedStats.getStdDeviation() : 0;
+        return extendedStats != null ? extendedStats.getStdDeviation() : Double.NaN;
     }
 
 
     public long getCardinality() {
-        return cardinality == null ? 0 : cardinality.getValue();
+        return cardinality != null ? cardinality.getValue() : Long.MIN_VALUE;
```

Comment:
I added the code to support that in here: https://github.com/Graylog2/graylog2-web-interface/commit/c89cd319f9d60582ef4297460db007de4e554c93